Description: Corrected XML conversion for style names, face name to avoid loading errors
Author: JS
Applied-Upstream: http://trac.wxwidgets.org/changeset/67390
Last-Update: 2012-05-31

Index: wxWidgets/branches/WX_2_8_BRANCH/src/richtext/richtextxml.cpp
===================================================================
--- a/src/richtext/richtextxml.cpp (revision 67247)
+++ b/src/richtext/richtextxml.cpp (revision 67390)
@@ -862,12 +862,12 @@
 
     wxString baseStyle = def->GetBaseStyle();
-    wxString baseStyleProp;
+    wxString baseStyleProp = wxT(" name=\"") + AttributeToXML(def->GetName()) + wxT("\"");
     if (!baseStyle.IsEmpty())
-        baseStyleProp = wxT(" basestyle=\"") + baseStyle + wxT("\"");
+        baseStyleProp = wxT(" basestyle=\"") + AttributeToXML(baseStyle) + wxT("\"");
 
     wxString descr = def->GetDescription();
     wxString descrProp;
     if (!descr.IsEmpty())
-        descrProp = wxT(" description=\"") + descr + wxT("\"");
+        descrProp = wxT(" description=\"") + AttributeToXML(descr) + wxT("\"");
 
     if (charDef)
@@ -896,5 +896,5 @@
 
         if (!listDef->GetNextStyle().IsEmpty())
-            baseStyleProp << wxT(" nextstyle=\"") << listDef->GetNextStyle() << wxT("\"");
+            baseStyleProp << wxT(" nextstyle=\"") << AttributeToXML(listDef->GetNextStyle()) << wxT("\"");
 
         OutputString(stream, wxT("<liststyle") + baseStyleProp + descrProp + wxT(">"), convMem, convFile);
@@ -937,5 +937,5 @@
 
         if (!paraDef->GetNextStyle().IsEmpty())
-            baseStyleProp << wxT(" nextstyle=\"") << paraDef->GetNextStyle() << wxT("\"");
+            baseStyleProp << wxT(" nextstyle=\"") << AttributeToXML(paraDef->GetNextStyle()) << wxT("\"");
 
         OutputString(stream, wxT("<paragraphstyle") + baseStyleProp + descrProp + wxT(">"), convMem, convFile);
@@ -991,5 +991,5 @@
 
         if (attr.HasFontFaceName())
-            str << wxT(" fontface=\"") << attr.GetFont().GetFaceName() << wxT("\"");
+            str << wxT(" fontface=\"") << AttributeToXML(attr.GetFont().GetFaceName()) << wxT("\"");
     }
 
@@ -1006,5 +1006,5 @@
 
     if (!attr.GetCharacterStyleName().empty())
-        str << wxT(" characterstyle=\"") << wxString(attr.GetCharacterStyleName()) << wxT("\"");
+        str << wxT(" characterstyle=\"") << AttributeToXML(attr.GetCharacterStyleName()) << wxT("\"");
 
     if (attr.HasURL())
@@ -1047,17 +1047,17 @@
                 str << wxT(" bulletsymbol=\"") << (int) (attr.GetBulletText()[0]) << wxT("\"");
             else
-                str << wxT(" bullettext=\"") << attr.GetBulletText() << wxT("\"");
-
-            str << wxT(" bulletfont=\"") << attr.GetBulletFont() << wxT("\"");
+                str << wxT(" bullettext=\"") << AttributeToXML(attr.GetBulletText()) << wxT("\"");
+
+            str << wxT(" bulletfont=\"") << AttributeToXML(attr.GetBulletFont()) << wxT("\"");
         }
 
         if (attr.HasBulletName())
-            str << wxT(" bulletname=\"") << attr.GetBulletName() << wxT("\"");
+            str << wxT(" bulletname=\"") << AttributeToXML(attr.GetBulletName()) << wxT("\"");
 
         if (!attr.GetParagraphStyleName().empty())
-            str << wxT(" parstyle=\"") << wxString(attr.GetParagraphStyleName()) << wxT("\"");
+            str << wxT(" parstyle=\"") << AttributeToXML(attr.GetParagraphStyleName()) << wxT("\"");
 
         if (!attr.GetListStyleName().empty())
-            str << wxT(" liststyle=\"") << wxString(attr.GetListStyleName()) << wxT("\"");
+            str << wxT(" liststyle=\"") << AttributeToXML(attr.GetListStyleName()) << wxT("\"");
 
         if (attr.HasTabs())
